### PR TITLE
Provide Spark 2.2 version project template for IntelliJ and Eclipse

### PR DESCRIPTION
**Pull request by t-rufang**

### CODE REVIEW

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/projects/SparkVersion.java [hunk 1]
```diff
@@ -25,6 +25,7 @@
 import java.util.Comparator;
 
 public enum SparkVersion {
+    SPARK_2_2_0("2.2.0", "2.11.8", "2.11"),
```
Comment (djlan) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/projects/SparkVersion.java]: >SPARK_2_2_0("2.2.0", "2.11.8", "2.11"), [](start = 4, length = 39)

Please do test all of our samples against Spark 2.2 cluster.

Comment (t-rufang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/projects/SparkVersion.java]: Ok, I will finish all the sample tests before I merge this PR.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/common2/SparkSubmitModel.java [hunk 2]
```diff
@@ -31,7 +31,9 @@
 import com.microsoft.azure.hdinsight.spark.common.SparkBatchSubmission;
 import com.microsoft.azure.hdinsight.spark.common.SparkSubmissionParameter;
 import com.microsoft.azure.hdinsight.spark.common.SparkSubmitResponse;
+import com.microsoft.azure.hdinsight.spark.jobs.JobUtils;
 import com.microsoft.tooling.msservices.components.DefaultLoader;
+import com.microsoft.tooling.msservices.helpers.CallableSingleArg;
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/common2/SparkSubmitModel.java]: Unused import?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/common2/SparkSubmitModel.java [hunk 3]
```diff
@@ -201,7 +203,8 @@ private void uploadFileToCluster(@NotNull final IClusterDetail selectedClusterDe
         
         String filePath = selectedClusterDetail.isEmulator() ?
                 SparkSubmitHelper.uploadFileToEmulator(selectedClusterDetail, buildJarPath) :
-                SparkSubmitHelper.uploadFileToHDFS(selectedClusterDetail, buildJarPath);
+                (selectedClusterDetail.getStorageAccount() == null ? SparkSubmitHelper.uploadFileToHDFS(selectedClusterDetail, buildJarPath) :
+                    JobUtils.uploadFileToAzure(new File(buildJarPath), selectedClusterDetail.getStorageAccount(), selectedClusterDetail.getStorageAccount().getDefaultContainerOrRootPath(), JobUtils.getFormatPathByDate(), HDInsightUtil.getToolWindowMessageSubject(), (CallableSingleArg)null));
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/common2/SparkSubmitModel.java]: `(CallableSingleArg)null` I suppose `null` is enough.

**Comment by wezhang** [comment]
fix #1583

**Pull request merged**